cmd/juju/application/deploy.go

```diff
@@ -794,9 +794,41 @@ func (c *DeployCommand) maybeReadLocalBundle() (deployFn, error) {
 	}, nil
 }
 
-func (c *DeployCommand) maybeReadLocalCharm() (deployFn, error) {
+func (c *DeployCommand) maybeReadLocalCharm(apiRoot DeployAPI) (deployFn, error) {
+	//NOTE: Here we select the series using the algorithm defined by the
+	//series selector. This serves to override the algorithm found in
+	//`charmrepo.NewCharmAtPath` which is outdated (but must still be called
+	//since the code is coupled with path interpretation logic which needs
+	//to be factored out).
+
+	ch, err := charm.ReadCharm(c.CharmOrBundle)
+	var series string
+	if err != nil {
+		series = c.Series
```

Comment:
Reading the charm here is only meant to aid in inferring the correct series, if this fails we just fall back to the argument series. If reading the charm fails here is will also fail below were it is handled properly. This is just an expedient to get the correct series. A proper refactoring of the `charmrepo` package is needed for a more elegant fix.